core/modules/query/QueryTemplate.cc

```diff
@@ -139,43 +100,52 @@ std::string QueryTemplate::sqlFragment() const {
     return oss.str();
 }
 
-// Output operator for QueryTemplate
+
 std::ostream& operator<<(std::ostream& os, QueryTemplate const& queryTemplate) {
-    SpacedOutput so(os);
-    std::for_each(queryTemplate._entries.begin(), queryTemplate._entries.end(), so);
+    std::string lastEntry;
+    std::string sep(" ");
+    for (auto const& entry : queryTemplate._entries ) {
+        std::string const& entryStr = entry->getValue();
+        if (entryStr.empty()) {
+            return os;
+        }
+        if (!lastEntry.empty()
+          && lsst::qserv::sql::sqlShouldSeparate(lastEntry, *lastEntry.rbegin(), entryStr.at(0))) {
```

Comment:
Probably a cut and paste from some other function. I'm not sure if there was a good reason for it.